mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -1761,4 +1772,111 @@ void CMomentumPlayer::PostThink()
     // Update previous origins
     NewPreviousOrigin(GetLocalOrigin());
     BaseClass::PostThink();
+}
+
+static float DamageForce(const Vector &size, float damage, float scale)
+{ 
+    float force = damage * ((48 * 48 * 82.0) / (size.x * size.y * size.z)) * scale;
+
+    if (force > 1000.0)
+    {
+        force = 1000.0;
+    }
+
+    return force;
+}
+
+int CMomentumPlayer::OnTakeDamage(const CTakeDamageInfo &info)
+{
+    CTakeDamageInfo adjustedInfo = info;
+
+    if (GetFlags() & FL_GODMODE)
+        return 0;
+
+    // Early out if there's no damage
+    if (!adjustedInfo.GetDamage())
+        return 0;
+
+    if (!IsAlive())
+        return 0;
+
+    CBaseEntity *pAttacker = adjustedInfo.GetAttacker();
+    CBaseEntity *pInflictor = adjustedInfo.GetInflictor();
+
+    // Take reduced damage midair from own rockets
+    if(pAttacker == this && FClassnameIs(pInflictor, "momentum_rocket") && GetGroundEntity() == NULL)
+    {
+        adjustedInfo.SetDamage(adjustedInfo.GetDamage() * mom_damagescale_self_rocket.GetFloat());
+    }
+
+    if (m_debugOverlays & OVERLAY_BUDDHA_MODE)
+    {
+        if (ceil(adjustedInfo.GetDamage()) >= m_iHealth)
+        {
+            m_iHealth = ceil(adjustedInfo.GetDamage()) + 1;
```

Comment:
Does health need to be modified at all? Re: your second issue, why not just remove this?